src/Microsoft.Graph.Core/Requests/Content/BatchRequestContent.cs

```diff
@@ -142,7 +146,8 @@ public string AddBatchRequestStep(IBaseRequest request)
                 });
 
             string requestId = Guid.NewGuid().ToString();
-            BatchRequestStep batchRequestStep = new BatchRequestStep(requestId, request.GetHttpRequestMessage());
+            var requestMessage = new HttpClientRequestAdapter(new AnonymousAuthenticationProvider()) {BaseUrl = baseClient.RequestAdapter.BaseUrl}.GetRequestMessageFromRequestInformation(requestInformation);
```

Comment:
since you already have a request adapter on the client, why not use that instead of creating a new instance?